src/jit/codegenxarch.cpp

```diff
@@ -5406,7 +5406,7 @@ void CodeGen::genJmpMethod(GenTreePtr jmp)
         // assert should hold.
         assert(varDsc->lvRegNum != REG_STK);
 
-        var_types loadType = varDsc->lvaArgType();
+        var_types loadType = genActualType(varDsc->lvaArgType()); // We own the memory and could use the full move.
```

Comment:
I think that this comment should be put with the comment above (and use "can" instead of "could"). Also, you should add an assert here that, if this is a field of a promoted struct, it is the only field (`lvIsStructField` is true for this, and `lvFieldCnt` is 1 for `lvParentLcl`). That way, if someone makes the change to promote struct parameters with multiple fields and misses the comment in lclVars.cpp, we are somewhat protected.